indico/web/client/js/react/components/notes/NoteConflictModal.jsx

```diff
@@ -0,0 +1,190 @@
+  Grid,
+  GridRow,
+  GridColumn,
+  List,
+  Segment,
+} from 'semantic-ui-react';
+
+import {Param, Translate} from 'indico/react/i18n';
+import {camelizeKeys} from 'indico/utils/case';
+
+import './NoteConflictModal.module.scss';
+
+export function NoteConflictModal({
+  currentNote,
+  externalNote,
+  onClose,
+  setCloseAndReload,
+  overwriteChanges,
+}) {
+  const [open, setOpen] = useState(true);
+
+  // camelize the externalNoteSource keys
+  const extNote = camelizeKeys(externalNote);
+
+  const renderDeletedRevision = () => {
+    return (
+      <Segment placeholder>
+        <Header icon>
+          <Icon name="trash outline alternate" />
+          <Translate>This note revision has been deleted.</Translate>
+        </Header>
+      </Segment>
+    );
+  };
+
+  return (
+    <Modal
+      open={open}
+      onClose={() => {
+        setOpen(false);
+        onClose();
+      }}
+      onOpen={() => setOpen(true)}
+      size="large"
+      style={{minWidth: '65vw'}}
+      closeOnEscape={false}
+      closeOnDimmerClick={false}
+      closeIcon
+    >
+      <Modal.Header>
+        <Translate>Conflict Detected</Translate>
+      </Modal.Header>
+      <Modal.Content>
+        <Message warning>
+          <Message.Header>
+            <Translate>This note has been modified by someone else</Translate>
+          </Message.Header>
+          <p>
+            <Translate>
+              This note was last modified on{' '}
+              <Param
+                name="last_modified_dt"
+                value={moment(extNote.createdDt).format('L LTS')}
+                wrapper={<strong />}
+              />{' '}
+              by <Param name="note_author" value={extNote.noteAuthor} wrapper={<strong />} />
+              {'. '}You can either:
+            </Translate>
+          </p>
+          <List style={{marginTop: '0.3rem'}}>
+            <List.Item>
+              <List.Icon name="write" />
+              <List.Content>
+                <Translate>Overwrite the changes made by the other person</Translate>
+              </List.Content>
+            </List.Item>
+            <List.Item>
+              <List.Icon name="trash alternate outline" />
+              <List.Content>
+                <Translate>Discard your changes</Translate>
+              </List.Content>
+            </List.Item>
+            <List.Item>
+              <List.Icon name="cancel" />
+              <List.Content>
+                <Translate>
+                  Close this message and return to the note without resolving the conflict
+                </Translate>
+              </List.Content>
+            </List.Item>
+          </List>
+        </Message>
+
+        <Grid divided="vertically" celled="internally">
+          <GridRow columns={2}>
+            <GridColumn>
+              <Header
+                as="h3"
+                content={Translate.string('Your changes')}
+                subheader={Translate.string('The changes you made to the note')}
```

Comment:
I'd delete this subheader since it doesn't add much useful info (and would make it more difficult to make this component reusable for the timetable)

alternatively it could be something more general, such as 'The changes you are attempting to save' or something like that

Reply:
I'm inclined to keep the header (for UI reasons), but I will generalise the wording of this :)